### DISCUSSION TITLE
Multiple Mixcoord components in a single deployment

### DISCUSSION THREAD

**Discussion by nairan-deshaw** [milvus-io/milvus#36544] [Q&A and General discussion] [answered]
Is it possible to have multiple replicas of a mixcoord in a single Milvus K8s cluster? We're seeing some issues in our mixcoord component and since there is only replica it becomes a single point of failure. When the replicas were increased to 2, the new pod keep crashing with `CompareAndSwap error for compare is false for key: querycoord`

**Comment by yhmo** [comment, marked as answer]
Milvus supports Coordinator HA from 2.3.3: https://milvus.io/docs/coordinator_ha.md#Coordinator-HA

Since the coordinators have been merged to mixcoord, we can config the value.yaml like this. And then increase the mixcoord k8s replicas to 2.
```
mixCoordinator:
  replicas: 2
  activeStandby:
    enabled: true
```

**Comment by nairan-deshaw** [comment]
When does the standby kick in? Is it only when k8s reports that the other node is not healthy or is there an internal health check between the two replicas?

**Reply by yhmo** [reply]
If the active coordinator is down, the standby coord will replace its work after a few seconds.

**Reply by nairan-deshaw** [reply]
We're facing an issue where the coordinator is timing out on all operations - https://github.com/milvus-io/milvus/issues/36217

The state shows okay, but none of the operations work. We wanted to know if this will be detected by the standby and use for automatic failover?
